### PR TITLE
chore: cleanup the jsoneditor file by splitting it into smaller reusa…

### PR DESCRIPTION
**Pull request by geoffreyoliaro**
CleanUp, master move the files into small reusable components(WIP)

### CODE REVIEW

File: src/lib/JSONEditor.js [hunk 1]
```diff
@@ -1,89 +1,28 @@
...
+import {
+  CollapseIcon,
+  isNodeCollapsed,
+  toggleNodeCollapsed,
+} from "./CollapseIcon";
+import {
+  styles,
+  EDIT_KEY,
+  Boolean,
+  getKey,
```
Comment (constantoduol) [inline src/lib/JSONEditor.js:20]: getKey, EDIT_KEY and styles are not components

File: src/lib/components/jsonEditorComponents/index.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export * from "./styles";
```
Comment (constantoduol) [inline src/lib/components/jsonEditorComponents/index.js:1]: this is unnecessary, remove this and make sure that imports can be imported as
```
import Boolean from './Boolean'
import ParentLabel from './ParentLabel'
```

File: src/lib/components/index.js [hunk 3]
```diff
@@ -0,0 +1 @@
+export * from "./jsonEditorComponents";
```
Comment (constantoduol) [inline src/lib/components/index.js:1]: there is no need for this, put all the components in the current components directory. There is no need to introduce another folder

Comment (geoffreyoliaro) [reply src/lib/components/index.js]: Resolved by using direct imports

File: src/lib/components/jsonEditorComponents/index.js (hunk 2 above)
Comment (geoffreyoliaro) [reply src/lib/components/jsonEditorComponents/index.js]: Resolved by using direct imports

File: src/lib/JSONEditor.js (hunk 1 above)
Comment (geoffreyoliaro) [reply src/lib/JSONEditor.js]: Added a helpers directory

**Comment by geoffreyoliaro** [comment]
Kindly review again

File: src/lib/JSONViewer.js [hunk 4]
```diff
@@ -1,14 +1,18 @@
 import React from 'react';
 import {isArray, isObject, isNumber, isString, isBoolean} from 'lodash';
 import {CollapseIcon, isNodeCollapsed, toggleNodeCollapsed} from './CollapseIcon';
+import { viewerStyles } from "./helpers/viewerStyles"
+import { getViewerKey } from "./helpers/ViewerKey"
+import { LabelAndValue } from './components/LabelAndValue'
+import { ViewerLabel } from './components/ViewerLabel'
```
Comment (constantoduol) [inline src/lib/JSONViewer.js:7]: why is this renamed?

File: src/lib/helpers/GetKey.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export const getKey = (prefix, currentKey, parentKeyPath, marginLeft) => {
```
Comment (constantoduol) [inline src/lib/helpers/GetKey.js:1]: just put all the helpers in one file, probably call it util.js or helpers.js

File: src/lib/helpers/viewerStyles.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
+export const viewerStyles = {
```
Comment (constantoduol) [inline src/lib/helpers/viewerStyles.js:1]: this can be renamed to jsonViewerDefaultStyles and the other styles to jsonEditorDefaultStyles. The styles can be overriden and these are just defaults

File: src/lib/helpers/styles.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
+export const styles = {
```
Comment (constantoduol) [inline src/lib/helpers/styles.js:1]: This can be renamed to jsonEditorDefaultStyles, see comment below

File: src/lib/JSONViewer.js (hunk 4 above)
Comment (geoffreyoliaro) [reply src/lib/JSONViewer.js:7]: There are two components called Label so I renamed the one associated to the viewerFile to be viewer label

File: src/lib/helpers/styles.js (hunk 7 above)
Comment (geoffreyoliaro) [reply src/lib/helpers/styles.js]: Resolved by renaming to jsonEditorDefaultStyles

File: src/lib/helpers/viewerStyles.js (hunk 6 above)
Comment (geoffreyoliaro) [reply src/lib/helpers/viewerStyles.js]: Resolved by renaming the files to jsonViewerDefaultStyles and jsonViewerDefaultStyles

File: src/lib/helpers/GetKey.js (hunk 5 above)
Comment (geoffreyoliaro) [reply src/lib/helpers/GetKey.js]: resolved by moving all the helpers to a util.js file

**Comment by constantoduol** [comment]
@geoffreyoliaro there is a change in this PR that breaks the JSON viewer, this is how it renders currently
React-App

This is how it should be

React-App (1)
